### ISSUE TITLE
another problem: Error: protect(): protection stack overflow

### ISSUE THREAD

**Issue by crazyhottommy** [kundajelab/phantompeakqualtools#3]
Hi, 

another problem with phantompeakqual or with R

```
Decompressing ChIP file
Decompressing control file
Loading required package: caTools
Reading ChIP tagAlign/BAM file Hs_940_temp//M940_Crep2.tagAlign.gz 
opened Hs_940_temp/M940_Crep2_spp_tmp//M940_Crep2.tagAlign144c7b15c3b9
done. read 42273553 fragments
ChIP data read length 37 
[1] TRUE
Reading Control tagAlign/BAM file Hs_940_.temp/control.tagAlign.gz 
opened Hs_940_.temp/M940_Crep2_spp_tmp//control.tagAlign144c62adab27
done. read 2922304 fragments
Error: protect(): protection stack overflow
Execution halted
Error: protect(): protection stack overflow
Execution halted
```

I googled and found http://stackoverflow.com/questions/28728774/how-to-set-max-ppsize-in-r

after adding one line 

`options("experssion" = 500000)`

on top of the run_spp.R script, same error.
Thanks for looking into this.
Tommy

**Comment by crazyhottommy** [comment]
any idea on this error?

**Comment by akundaje** [comment]
Not sure. Haven't come across this ever before. Have you tried our
automated pipeline https://github.com/kundajelab/chipseq_pipeline

Anshul

**Comment by crazyhottommy** [comment]
I actually wrote a snakemake version of the pipeline to accommodate our own usage. I know the bolts and nuts of my own pipeline so I can  easily extend it.

I will continue to debug...thanks though.

Tommy

**Comment by leepc12** [comment]
http://stackoverflow.com/questions/28728774/how-to-set-max-ppsize-in-r

Can you try with higher max stack size "Rscript --max-ppsize=500000
[RFILE]"? It's 5000 by default and should be between 5000 and 5000000.

Thanks,

Jin

**Comment by crazyhottommy** [comment]
@leepc12 I added `options("experssion" = 500000)` to the run_spp.R script, but still same error. puzzling.

**Comment by andreamariossi** [comment]
Hi,
having the same problem. I tried the suggested solutions `Rscript --max-ppsize=500000
[RFILE]` or adding `options("exppression" = 500000) `. I tried both on Xubuntu 16.04 LTS and Red Hat 4.4.7-3.
thanks,
Andrea

**Comment by seenstevo** [comment]
@leepc12 
Hi,
I have run into the same problem recently while working with DNase-seq on a new species. I know the program is still working as old tagAlign files are processed fine. My working theory is that it may have something to do with the number of scaffolds/chromosomes (col 1). The reason is that this new species currently has a basic genome assembly so there are many very small scaffolds (something like >200000). I tested two files derived from one file that fails with "Error: protect(): protection stack overflow", one was produced from `head -100000` and one from `tail -100000`. The head file worked fine while the tail one failed with the same error (Error: protect(): protection stack overflow). Given that they are the same size etc and the scaffolds are ordered by size (largest first), the head derived file only has one or two different scaffold/chr while the tail derived file has a few rows per scaffold/chr so has thousands of different ones so this is what I thought might be the culprit.

@andreakcl89 , @crazyhottommy Does this fit with your issue? Has anyone found a fix?
Thanks, Sean

**Comment by oshomroni** [comment]
I have the exact same issue using ATAC_seq workflow provided by the Kundaje lab. I ran it for Xenopus laevis version 9.2, which contains 18 chromosomes and 108015 scaffolds. In terms of proportions, non-mitochondrial chromosomes cover 92% of the genome, a very small percentage is mitochondrial DNA, and 7.7% of the genome is scaffolds. If, as @seenstevo mentioned, the issue is the massive number of scaffolds that the analysis has to run through, maybe it is not too bad to remove 7.7% of the genome to get the analysis done

**Comment by akundaje** [comment]
The cross correlation analysis is not necessary for ATAC seq data. So you
should simply disable that.

Jin - is there a parameter to disable CC analysis for ATAC/DNase

**Comment by crazyhottommy** [comment]
@seenstevo no, I am processing the same human data. somehow in a different computing cluster, the error disappeared.

**Comment by oshomroni** [comment]
I added "Rscript --max-ppsize=500000" inside the postalign_xcor.bds script before ${RUN_SPP}, and it worked. Guess you need a good server with tons of memory for this to work.

**Comment by leepc12** [comment]
You can disable cross correlation analysis by activating pipeline's flag
`-no_xcor​`.

Jin
